rend-spec-v3.txt

```diff
@@ -2334,31 +2324,51 @@ Appendix E. Managing authorized client data [CLIENT-AUTH-MGMT]
      human-readable authorized client names.
 
      Let's consider that one of the listed client names is "alice". In this
-     case, Tor checks the "client_authorized_pubkeys" file for any entries
-     with client_name being "alice". If an "alice" entry is found, we use the
-     relevant pubkeys to authenticate Alice.
+     case, Tor checks whether the "./client_authorized_privkeys/alice.keys"
+     file is present in the filesystem:
 
-     If no "alice" entry is found in the "client_authorized_pubkeys" file, Tor
-     is tasked with generating public/private keys for Alice. To do so, Tor
-     generates x25519 and ed25519 keypairs for Alice, then makes a
-     "client_authorized_privkeys/alice.privkey" file and writes the private
-     keys inside; it also adds an entry for alice to the
-     "client_authorized_pubkeys" file.
+     If the "alice.keys" file is found, we use the included keys to
+     authenticate Alice.
+
+     If no "alice.keys" file is found, Tor is tasked with generating Alice's
+     keypair. To do so, Tor generates x25519 and ed25519 keypairs for Alice,
```

Comment:
Does the service need to generate both keypairs? What if that client is enabled in desc auth but not in intro auth (via HiddenServiceAuthorizeClient)?

Reply:
In that case the service would need to indeed generate only the desc key.

I didn't end up describing the standard/desc/intro auth distinction in the spec patch because it would get too complex. Perhaps we should do it before proceeding tho.